File: producer/nim/fibonacci.nim
import bigints

proc fib() =
  let filename = "fibonacci.txt"
  var file: File
  try:
    file = open(filename, fmWrite)
  except IOError:
    stderr.writeLine("Error opening file '" & filename & "'")
    quit(1)
  defer:
    file.close()

  const CHUNK_SIZE = 10_000
  var a: BigInt = 0.initBigInt
  var b: BigInt = 1.initBigInt
  var n: int = 0
  while true:
    for i in 0 ..< CHUNK_SIZE:
      file.writeLine(a)
      let c = a + b
      a = b
      b = c
    file.flushFile()
    n += CHUNK_SIZE
    echo n, " total numbers written."

fib()

File: producer/cpp/fibonacci.cpp
#include <fstream>
#include <iostream>
#include <string>

#include <boost/multiprecision/cpp_int.hpp>

namespace mp = boost::multiprecision;

typedef mp::cpp_int bigint;

int main([[maybe_unused]] int argc, [[maybe_unused]] char* argv[]) {
  std::string filename = "fibonacci.txt";
  std::ofstream file(filename);
  if (!file) {
    std::cerr << "Error opening file '" << filename << "'" << std::endl;
    return EXIT_FAILURE;
  }
  bigint a = 0;
  bigint b = 1;
  std::size_t n = 0;
  constexpr std::size_t CHUNK_SIZE = 10'000;
  while (true) {
    for (std::size_t i = 0; i < CHUNK_SIZE; ++i) {
      file << a << '\n';
      bigint c = a + b;
      a = b;
      b = c;
    }
    file.flush();
    n += CHUNK_SIZE;
    std::cout << n << " total numbers written." << std::endl;
  }
  return EXIT_SUCCESS;
}
